x-pack/solutions/search/plugins/enterprise_search/server/deprecations/pre_eight_index_deprecator.ts

```diff
@@ -37,8 +44,28 @@ export async function getPreEightEnterpriseSearchIndices(
       const dataStreamName = indexData.data_stream;
       returnIndices.push({
         name: index,
-        isDatastream: dataStreamName ? true : false,
-        datastreamName: dataStreamName ?? '',
+        hasDatastream: dataStreamName ? true : false,
+        datastreams: [dataStreamName ?? ''],
+      });
+    }
+  }
+
+  for (const [datastream, datastreamData] of Object.entries(entSearchDatastreams)) {
+    if (!datastreamData.indices || datastreamData.indices.length === 0) {
+      continue;
+    }
+
+    // only get the last index, as this the current write index
+    const lastIndexName = datastreamData.indices[datastreamData.indices.length - 1].index_name;
+    const existingIndex = returnIndices.find((index) => index.name === lastIndexName);
+    if (existingIndex) {
+      existingIndex.hasDatastream = true;
+      existingIndex.datastreams.push(datastream);
+    } else {
+      returnIndices.push({
+        name: lastIndexName,
+        hasDatastream: true,
+        datastreams: [datastream],
```

Comment:
I think this makes sense: if the index is already picked up we make sure it's marked as having a data stream. Otherwise it's not been picked up and we add to the list of indices to make read-only.